### PR TITLE
Map and plant

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
Moves both the rat king and mouse in the mines to the medical area
Adds a new potted plant sprited by PolyNayu

### CODE REVIEW

File: code/game/objects/structures/flora/flora_potted.dm [hunk 1]
```diff
@@ -297,3 +297,35 @@
 	desc = "This is a tiny well lit decorative christmas tree."
 	icon_state = "plant-xmas"
 	needs_to_maintain = TRUE
+
+//Shard
+/obj/structure/flora/pottedplant/green_rock
+	name = "contained ameridian shard"
+	desc = "A contained shiney shard of ameridian, uses a posi sound for its power and has some osium for emp shielding making it super safe.."
```
Comment (r4d6) [inline code/game/objects/structures/flora/flora_potted.dm:304]:
```suggestion
	desc = "A contained shiney shard of ameridian, uses a posi sound for its power and has some osium for emp shielding making it super safe.."
```
```suggestion
	desc = "A contained shiny shard of ameridian, uses a positronic cell for its power and has some osium for emp shielding making it super safe."
```

File: code/game/objects/structures/flora/flora_potted.dm [hunk 2]
```diff
@@ -297,3 +297,35 @@
...
+	var/shield_level = 3
+
+/obj/structure/flora/pottedplant/green_rock/attack_hand(mob/living/user as mob)
+	if(shield_level >= 3)
+		shield_level -= 1
+	if(shield_level <= -1)
+		shield_level += 1
+	if(shield_level == 3)
+		icon_state = "ameridian_pot_shield_powered"
+		desc = "A contained shiney shard of ameridian, uses a posi sound for its power and has some osium for emp shielding making it super safe.."
```
Comment (r4d6) [inline code/game/objects/structures/flora/flora_potted.dm:321]:
```suggestion
		desc = initial(desc)
```

File: code/game/objects/structures/flora/flora_potted.dm [hunk 3]
```diff
@@ -297,3 +297,35 @@
...
+		return
+	if(shield_level == 2)
+		icon_state = "ameridian_pot_shield_unpowered"
+		desc = "A contained shiney shard of ameridian, uses a posi sound for its power and has some osium for emp shielding making it super safe.."
```
Comment (r4d6) [inline code/game/objects/structures/flora/flora_potted.dm:325]:
```suggestion
		desc = initial(desc)
```

File: code/game/objects/structures/flora/flora_potted.dm [hunk 4]
```diff
@@ -297,3 +297,35 @@
...
+		return
+	if(shield_level == 1)
+		icon_state = "ameridian_pot_shieldless"
+		desc = "A contained shiney shard of ameridian, uses a posi sound for its power and has some osium for emp shielding making it super safe. Its shield light level is set to its lowest level making it still contained but look nice."
```
Comment (r4d6) [inline code/game/objects/structures/flora/flora_potted.dm:329]:
```suggestion
		desc = initial(desc)
		desc += " Its shield light level is set to its lowest level, making it still contained but look nice."
```

File: code/game/objects/structures/flora/flora_potted.dm [hunk 5]
```diff
@@ -297,3 +297,35 @@
...
+		return
+	shield_level = 3 //Reset out of bound list
```
Comment (r4d6) [inline code/game/objects/structures/flora/flora_potted.dm:331]:
```suggestion
	shield_level = 3 //Reset out of bound list

```

File: code/game/objects/structures/flora/flora_potted.dm [hunk 6]
```diff
@@ -297,3 +297,35 @@
...
+	icon_state = "ameridian_pot_shield_powered"
+	layer = PROJECTILE_HIT_THRESHHOLD_LAYER
+	needs_to_maintain = FALSE
+	light_range = 2
+	light_power = 0.6
+	light_color = "#43BA69" //Green!
+	var/shield_level = 3
+
+/obj/structure/flora/pottedplant/green_rock/attack_hand(mob/living/user as mob)
+	if(shield_level >= 3)
```
Comment (r4d6) [inline code/game/objects/structures/flora/flora_potted.dm:315]: Using a switch statement for the whole thing might be more efficient.

Comment (Trilbyspaceclone) [reply code/game/objects/structures/flora/flora_potted.dm:315]: prob but i dont know how to do swap statements

File: code/game/objects/structures/flora/flora_potted.dm (hunk 2 above)
Comment (Trilbyspaceclone) [reply code/game/objects/structures/flora/flora_potted.dm:321]: Doing it like this do to it being less memory heavy

File: code/game/objects/structures/flora/flora_potted.dm (hunk 6 above)
Comment (r4d6) [reply code/game/objects/structures/flora/flora_potted.dm:315]: http://www.byond.com/docs/ref/#/proc/switch

File: code/game/objects/structures/flora/flora_potted.dm (hunk 3 above)
Comment (Trilbyspaceclone) [reply code/game/objects/structures/flora/flora_potted.dm:325]: Less memory intensive this way

File: code/game/objects/structures/flora/flora_potted.dm (hunk 4 above)
Comment (Trilbyspaceclone) [reply code/game/objects/structures/flora/flora_potted.dm:329]: Less memory intensive this way

**Pull request merged**